### PR TITLE
bank: add BankService definitions

### PR DESCRIPTION
**Pull request by jwangsadinata**
This commit adds the bank service definitions, which include CreateBank,
GetBank, and CreateTransactionTypeAtBank.

### CODE REVIEW

File: v1/bank/bank.gunk [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+	// +gunk openapiv2.Operation{
+	//         Tags:        []string{"Bank"},
+	//         Description: "Creates a new bank and returns its id.",
+	//         Summary:     "Create a bank",
+	//         Responses: map[string]openapiv2.Response{
+	//                 "201": openapiv2.Response{
+	//                         Description: "Bank created successfully.",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/bankCreateBankResponse",
```
Comment (ghazninattarshah) [inline v1/bank/bank.gunk]: this would just be `/bankBank`, as there is no CreateBankResponse

File: v1/bank/bank.gunk [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+	// +gunk openapiv2.Operation{
+	//         Tags:        []string{"Bank"},
+	//         Description: "Creates a new transaction type at a bank and returns its transaction type response.",
+	//         Summary:     "Create a bank",
+	//         Responses: map[string]openapiv2.Response{
+	//                 "201": openapiv2.Response{
+	//                         Description: "TransactionType created successfully.",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/bankCreateBankResponse",
```
Comment (ghazninattarshah) [inline v1/bank/bank.gunk]: /bankTransactionType

File: v1/bank/messages.pot [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		"Report-Msgid-Bugs-To: %s\n"
+		"POT-Creation-Date: %s\n"
+		"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+		"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+		"Language-Team: LANGUAGE <LL@li.org>\n"
+		"Language: \n"
+		"MIME-Version: 1.0\n"
+		"Content-Type: text/plain; charset=CHARSET\n"
+		"Content-Transfer-Encoding: 8bit\n"
+
```
Comment (ghazninattarshah) [inline v1/bank/messages.pot:18]: Is this an auto-generated file as well? the values are missing !!

**Review by ghazninattarshah — APPROVED** [review]
LGTM.

File: v1/bank/bank.gunk [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+	// NationalIdentifier is the national identifier code.
+	NationalIdentifier string `pb:"7" json:"national_identifier"`
+
+	// BankRouting is the routing information for the bank.
+	BankRouting BankRouting `pb:"8" json:"bank_routing"`
+}
+
+// TransactionType is holds the information about transaction type at a bank.
+type TransactionType struct {
+}
```
Comment (favadi) [inline v1/bank/bank.gunk:119]: why this struct is empty?

File: v1/bank/bank.gunk [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+	ID string `pb:"1" json:"id"`
+
+	// FullName is the full name of the bank.
+	FullName string `pb:"2" json:"full_name"`
+
+	// ShortName is the short name of the bank.
+	ShortName string `pb:"3" json:"short_name"`
+
+	// LogoURL is the url for the bank's logo.
+	LogoURL string `pb:"4" json:"logo_url"`
```
Comment (favadi) [inline v1/bank/bank.gunk]: Do we really need to search for bank by logo URL? :P

Comment (nkristianto) [reply v1/bank/bank.gunk]: ahh, agree with @favadi 
i don't think this necessary to put LogoURL

File: v1/bank/bank.gunk [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+type GetBanksResponse struct {
+	// Banks is the list of the banks.
+	Banks []Bank `pb:"1" json:"banks"`
+}
+
+// CreateTransactionTypeAtBankRequest is the request envelope to create a new
+// transaction type.
+type CreateTransactionTypeAtBankRequest struct {
+	// TransactionType is the related information about a transaction type at a bank.
+	TransactionType TransactionType `pb:"1" json:"transaction_type"`
```
Comment (nkristianto) [inline v1/bank/bank.gunk:164]: why we have TransactionType as a request and response (L354)

File: v1/bank/bank.gunk (hunk 1 above)
Comment (jwangsadinata) [reply v1/bank/bank.gunk]: great catch.

File: v1/bank/bank.gunk (hunk 2 above)
Comment (jwangsadinata) [reply v1/bank/bank.gunk]: thanks ghazni.

File: v1/bank/messages.pot (hunk 3 above)
Comment (jwangsadinata) [reply v1/bank/messages.pot:18]: this is autogenerated, but I'll try again on the incoming fix.

File: v1/bank/bank.gunk (hunk 5 above)
Comment (jwangsadinata) [reply v1/bank/bank.gunk]: Good catch, I was copying from the payload based on https://apiexplorer.openbankproject.com/?version=OBPv4.0.0&list-all-banks=false&core=&psd2=&obwg=&ignoredefcat=&tags=&bank_id=aachener-bank-aachen#OBPv3_0_0-bankById, which specify logoURL as a search parameter but it really does not make sense to have this. Thanks for the review guys.

File: v1/bank/bank.gunk (hunk 6 above)
Comment (jwangsadinata) [reply v1/bank/bank.gunk:164]: Ah, on the request, we are sending this is the body payload, and what is being returned as a response is the same body that we have created through the endpoint.

File: v1/bank/bank.gunk (hunk 4 above)
Comment (jwangsadinata) [reply v1/bank/bank.gunk:119]: good catch, I forgot to put the payload on this. 🙈

**Comment by jwangsadinata** [comment]
@nkristianto @favadi, updated accordingly, can you guys help in taking another look at this?

File: v1/bank/bank.gunk (hunk 6 above)
Comment (ghazninattarshah) [reply v1/bank/bank.gunk:164]: I think it's just going to return what's saved with id.

Comment (nkristianto) [reply v1/bank/bank.gunk:164]: agree with ghazni

Comment (nkristianto) [reply v1/bank/bank.gunk:164]: i think we should have some standard to return create operation, maybe with ID. For the case that needs more than ID as a return value, we can customize that later.

**Review by nkristianto — APPROVED** [review]

**Pull request merged**